trace/amqp/integration_test.go

```diff
@@ -0,0 +1,77 @@
+// !build integration
+
+package amqp
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPublisherSuccess(t *testing.T) {
+	ctx := context.Background()
```

Comment:
we should test that a message was actually published and received.

Reply:
Done; I created a small consumer which tests that the message was received correctly. I came across issues of messages spilling between different this and async/amqp tests, so I separated them into different queues.